src/DotNet.Glob/Evaluation/LiteralTokenEvaluator.cs

```diff
@@ -20,6 +22,12 @@ public bool IsMatch(string allChars, int currentPosition, out int newPosition)
                 var compareChar = _token.Value[counter];
                 var currentChar = allChars[newPosition];
 
+                if (_caseInsensitive)
+                {
+                    compareChar = char.ToLowerInvariant(compareChar);
```

Comment:
Same here.. I am thinking _token.Value[] should probably be stored in a LowerInvariant() form at parse time, and then IsMatch wouldn't need to do that at Match time for each match.

Reply:
That makes sense. ToLowerInvariant is a sufficiently expensive operation to try to avoid it as much as possible.